File: exercism/python/pangram/pangram.py
#!/usr/bin/python3
# https://exercism.io/my/solutions/fb0fec300bdc4f5e81409cf243f96b12


from string import ascii_lowercase


ALPHABET_SET = set(ascii_lowercase)


def is_pangram(sentence):
    sentence_set = set(filter(lambda c: c in ascii_lowercase, str.lower(sentence)))
    return sentence_set == ALPHABET_SET

File: exercism/go/pangram/pangram.go
// https://exercism.io/my/solutions/8de4b7fc7e534cefa397ce9fa340b513

package pangram

import "unicode"

// IsPangram determines if every letter in an alphabet appears or not in a sentence.
func IsPangram(sentence string) bool {
	const alphabet = "abcdefghijklmnopqrstuvwxyz"

	present := make(map[rune]bool, len(alphabet))
	for _, letter := range alphabet {
		present[letter] = false
	}

	for _, letter := range sentence {
		present[unicode.ToLower(letter)] = true
	}

	for _, v := range present {
		if !v {
			return false
		}
	}
	return true
}
